### ISSUE TITLE
"type" field not respected during publish

### ISSUE THREAD

**Issue by ImLunaHey** [onlxltd/bonjour-service#27]
Should `type` not be set to `floopy-http`?

```js
// @ts-check
const { hostname } = require('os');
const { Bonjour } = require('bonjour-service');

const bonjour = new Bonjour();

bonjour.publish({
  type: 'floopy-http',
  name: `Floopy-${hostname()}`,
  port: 3000,
});

bonjour.find({
  type: 'floopy-http'
}, service => {
  console.log(service);
//   {
//     addresses: [],
//     name: 'Floopy-Sophie-2',
//     fqdn: 'Floopy-Sophie-2.local._floopy-http._tcp.local',
//     host: 'Sophie-2.local',
//     referer: { address: '192.168.1.163', family: 4, port: 5353, size: 230 },
//     port: 3000,
//     type: 'local',
//     protocol: 'floopy-http',
//     subtypes: [ 'tcp' ],
//     rawTxt: [],
//     txt: {}
//   }
});
```

**Comment by ImLunaHey** [comment]
Seems if `name` ends with a `.something` it'll the part after the `.` as the type.

```js
// @ts-check
const { Bonjour } = require('bonjour-service');

const service = new Bonjour();
const browser = service.find({ type: 'floopy-http' });

browser.on('up', (service) => {
  console.log('up', service)
//   up {
//     addresses: [],
//     name: 'name',
//     fqdn: 'name.this-is-broken._floopy-http._tcp.local',
//     host: 'Sophie-2.local',
//     referer: { address: '192.168.1.163', family: 4, port: 5353, size: 224 },
//     port: 3000,
//     type: 'this-is-broken',
//     protocol: 'floopy-http',
//     subtypes: [ 'tcp' ],
//     rawTxt: [],
//     txt: {}
//   }
//   up {
//     addresses: [],
//     name: 'name-this-is-okay',
//     fqdn: 'name-this-is-okay._floopy-http._tcp.local',
//     host: 'Sophie-2.local',
//     referer: { address: '192.168.1.163', family: 4, port: 5353, size: 218 },
//     port: 3000,
//     type: 'floopy-http',
//     protocol: 'tcp',
//     subtypes: [],
//     rawTxt: [],
//     txt: {}
//   }
});

service.publish({ name: `name.this-is-broken`, type: 'floopy-http', port: 3000 });
service.publish({ name: `name-this-is-okay`, type: 'floopy-http', port: 3000 });
```

**Comment by mdidon** [comment]
@ImLunaHey Thanks for raising the issue. It looks like this relates to the way the name and fqdn are configured in service.ts. We'll add this for review in the next release.

**Comment by mdidon** [comment]
Merged in #29
